tests/integration/create_pro_user.t

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/perl -w
+
+use ProductOpener::PerlStandards;
+
+use ProductOpener::Config qw/$data_root/;
+use ProductOpener::Store qw/retrieve/;
+
+use Test::More;
+use ProductOpener::APITest qw/:all/;
+use ProductOpener::Test qw/:all/;
+use ProductOpener::TestDefaults qw/:all/;
+
+use Clone qw/clone/;
+use List::Util qw/first/;
+use Storable qw(dclone);
+
+=head1 Test creation of a producer user
+
+=cut
+
+my ($test_id, $test_dir, $expected_result_dir, $update_expected_results) = (init_expected_results(__FILE__));
+
+# clean
+remove_all_users();
+remove_all_orgs();
+wait_application_ready();
+
+my $admin_ua = new_client();
+my $resp = create_user($admin_ua, \%admin_user_form);
+ok(!html_displays_error($resp));
+# create a pro moderator
+my $moderator_ua = new_client();
+$resp = create_user($moderator_ua, \%pro_moderator_user_form);
+ok(!html_displays_error($resp));
+# admin pass him moderator
+my %moderator_edit_form = (
+	%pro_moderator_user_form,
+	user_group_pro_moderator => "1",
+	type => "edit",
+);
+$resp = edit_user($admin_ua, \%moderator_edit_form);
+ok(!html_displays_error($resp));
+
+# A user comes along, creates its profile and requests to be part of an org
+my $user_ua = new_client();
+my %user_form = (
+	%{clone(\%default_user_form)},
+	pro => "1",
+	requested_org => "Acme Inc."
+);
+my $tail = tail_log_start();
+$resp = create_user($user_ua, \%user_form);
+ok(!html_displays_error($resp));
+my $logs = tail_log_read($tail);
+
+# As it is the first user of the org, user is already part of the org
+my $user_ref = retrieve("$data_root/users/tests.sto");
+# user is already part of org
+is($user_ref->{pro}, 1);
+
+is($user_ref->{org}, "acme-inc");
+is($user_ref->{org_id}, "acme-inc");
+# remove password for comparison and timestamps
+my $user_cmp_ref = dclone($user_ref);
+normalize_user_for_test_comparison($user_cmp_ref);
+compare_to_expected_results($user_cmp_ref, "$expected_result_dir/user-after-subscription.json",
+	$update_expected_results);
+
+# Org was created
+my $org_ref = retrieve("$data_root/orgs/acme-inc.sto");
+my $org_cmp_ref = dclone($org_ref);
+normalize_org_for_test_comparison($org_cmp_ref);
+compare_to_expected_results($org_cmp_ref, "$expected_result_dir/org-after-subscription.json", $update_expected_results);
+
+# Get mails from log
+my @mails = mails_from_log($logs);
+# we got three
+is(scalar @mails, 3);
```

Comment:
I think we could replace all the checks on the emails below by one line:

compare_to_expected_results(\@mails, "$expected_result_dir/mails.json", $update_expected_results);